src/backend/replication/logical/logical.c

```diff
@@ -137,7 +137,9 @@ StartupDecodingContext(List *output_plugin_options,
 	 * (re-)load output plugins, so we detect a bad (removed) output plugin
 	 * now.
 	 */
-	LoadOutputPlugin(&ctx->callbacks, NameStr(slot->data.plugin));
+
+	if (strlen(slot->data.plugin.data))
```

Comment:
Why is this change needed? Why would a slot not have a plugin registered? 

`if (strlen(NameStr(slot->data.plugin)))`